src/home/homectl.c

```diff
@@ -4430,54 +4452,25 @@ static int parse_argv(int argc, char *argv[]) {
                         break;
                 }
 
-                case ARG_TASKS_MAX: {
-                        uint64_t u;
 
-                        if (isempty(optarg)) {
-                                r = drop_from_identity("tasksMax");
-                                if (r < 0)
-                                        return r;
-                                break;
-                        }
-
-                        r = safe_atou64(optarg, &u);
-                        if (r < 0)
-                                return log_error_errno(r, "Failed to parse --tasks-max= parameter: %s", optarg);
-
-                        r = sd_json_variant_set_field_unsigned(match_identity ?: &arg_identity_extra, "tasksMax", u);
+                case ARG_TASKS_MAX:
+                        r = parse_u64_field(match_identity ?: &arg_identity_extra, "tasksMax", optarg);
                         if (r < 0)
-                                return log_error_errno(r, "Failed to set tasksMax field: %m");
-
+                                return r;
                         break;
-                }
 
                 case ARG_MEMORY_MAX:
                 case ARG_MEMORY_HIGH:
                 case ARG_LUKS_PBKDF_MEMORY_COST: {
                         const char *field =
                                             c == ARG_MEMORY_MAX ? "memoryMax" :
                                            c == ARG_MEMORY_HIGH ? "memoryHigh" :
-                                c == ARG_LUKS_PBKDF_MEMORY_COST ? "luksPbkdfMemoryCost" : NULL;
+                                c == ARG_LUKS_PBKDF_MEMORY_COST ? "luksPbkdfMemoryCost"
+                                                                : NULL;
 
-                        uint64_t u;
-
-                        assert(field);
-
-                        if (isempty(optarg)) {
-                                r = drop_from_identity(field);
-                                if (r < 0)
-                                        return r;
-                                break;
-                        }
-
-                        r = parse_size(optarg, 1024, &u);
-                        if (r < 0)
-                                return log_error_errno(r, "Failed to parse %s parameter: %s", field, optarg);
-
-                        r = sd_json_variant_set_field_unsigned(match_identity ?: &arg_identity_extra_this_machine, field, u);
+                        r = parse_u64_field(match_identity ?: &arg_identity_extra_this_machine, field, optarg);
```

Comment:
This is wrong, `parse_size()` does a lot more than `parse_u64_field()` does. Please add a helper that uses `parse_size()`